near-sdk-macros/src/core_impl/abi/abi_generator.rs

```diff
@@ -126,6 +127,34 @@ impl ImplItemMethodInfo {
                         None
                     }
                 }
+                ReturnType::Type(_, ty) if utils::type_is_result(ty) && is_handles_result => {
```

Comment:
nit: not sure if rust reorders the conditional so that we can short circuit earlier with just `is_handles_result` instead of evaluating via `type_is_result(ty)`
```suggestion
                ReturnType::Type(_, ty) if is_handles_result && utils::type_is_result(ty) => {
```

Reply:
Not certain about this, seems it isn't guaranteed from https://github.com/rust-lang/rust/issues/83623 and related things. Prob best to re-order to be safe, but this additional computation only happens at compile time to definitely a micro-optimization